specification/azurefleet/AzureFleet.Management/baseVirtualMachineProfile.tsp

```diff
@@ -0,0 +1,2518 @@
+   * scale set.
+   */
+  securityProfile?: SecurityProfile;
+
+  /**
+   * Specifies the boot diagnostic settings state. Minimum api-version: 2015-06-15.
+   */
+  diagnosticsProfile?: DiagnosticsProfile;
+
+  /**
+   * Specifies a collection of settings for extensions installed on virtual machines
+   * in the scale set.
+   */
+  extensionProfile?: VirtualMachineScaleSetExtensionProfile;
+
+  /**
+   * Specifies that the image or disk that is being used was licensed on-premises.
+   * <br><br> Possible values for Windows Server operating system are: <br><br>
+   * Windows_Client <br><br> Windows_Server <br><br> Possible values for Linux
+   * Server operating system are: <br><br> RHEL_BYOS (for RHEL) <br><br> SLES_BYOS
+   * (for SUSE) <br><br> For more information, see [Azure Hybrid Use Benefit for
+   * Windows
+   * Server](https://docs.microsoft.com/azure/virtual-machines/windows/hybrid-use-benefit-licensing)
+   * <br><br> [Azure Hybrid Use Benefit for Linux
+   * Server](https://docs.microsoft.com/azure/virtual-machines/linux/azure-hybrid-benefit-linux)
+   * <br><br> Minimum api-version: 2015-06-15
+   */
+  licenseType?: string;
+
+  /**
+   * Specifies Scheduled Event related configurations.
+   */
+  scheduledEventsProfile?: ScheduledEventsProfile;
+
+  /**
+   * UserData for the virtual machines in the scale set, which must be base-64
+   * encoded. Customer should not pass any secrets in here. Minimum api-version:
+   * 2021-03-01.
+   */
+  userData?: string;
+
+  /**
+   * Specifies the capacity reservation related details of a scale set. Minimum
+   * api-version: 2021-04-01.
+   */
+  capacityReservation?: CapacityReservationProfile;
+
+  /**
+   * Specifies the gallery applications that should be made available to the VM/VMSS
+   */
+  applicationProfile?: ApplicationProfile;
+
+  /**
+   * Specifies the hardware profile related details of a scale set. Minimum
+   * api-version: 2021-11-01.
+   */
+  hardwareProfile?: VirtualMachineScaleSetHardwareProfile;
+
+  /**
+   * Specifies the service artifact reference id used to set same image version for
+   * all virtual machines in the scale set when using 'latest' image version.
+   * Minimum api-version: 2022-11-01
+   */
+  serviceArtifactReference?: ServiceArtifactReference;
+
+  /**
+   * Specifies the security posture to be used for all virtual machines in the scale
+   * set. Minimum api-version: 2023-03-01
+   */
+  securityPostureReference?: SecurityPostureReference;
+
+  /**
+   * Specifies the time in which this VM profile for the Virtual Machine Scale Set
+   * was created. Minimum API version for this property is 2023-09-01. This value
+   * will be added to VMSS Flex VM tags when creating/updating the VMSS VM Profile
+   * with minimum api-version 2023-09-01.
+   */
+  @visibility("read")
+  // FIXME: (utcDateTime) Please double check that this is the correct type for your scenario.
+  timeCreated?: utcDateTime;
```

Comment:
I see a FIXME warning here.
Please verify you can remove it

Reply:
Updated